### PR TITLE
PR Template: Add test related checkboxes

### PR DESCRIPTION
**Pull request by brbrr**
Adds checkboxes to the PR template to nudge folks to add tests to their changes as well as to check the results of E2E tests. Inspired by Woo template.

#### Jetpack product discussion

n/a
#### Does this pull request change what data or activity we track or use?

n/a
#### Testing instructions:

* Does the wording make sense?
* Is there anything else we can ask folks?

The checklist would look like this:

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes does not break them?

### CODE REVIEW

**Review by jeherve — COMMENTED** [review]
I like it! 

I think I would move that section a bit higher up though, right under "Changes proposed in this Pull Request". In my opinion those are questions you must ask yourself before to even start thinking about testing instructions for the PR reviewer.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -26,3 +26,8 @@ Fixes #
 
 * Go to '..'
 *
+
+### Other information:
+
+- [ ] Have you written new tests for your changes, if applicable?
+- [ ] Have you checked the E2E test CI results, and verified that your changes does not break them?
```
Comment (manzoorwanijk) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
```

**Review by manzoorwanijk — DISMISSED** [review]
Nice change. Needs just a typo fix.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (adimoldovan) [inline .github/PULL_REQUEST_TEMPLATE.md]: I might re-word it in the first person, as long as there are check boxes to tick.

```suggestion
- [ ] I have written new tests for my changes / tests are not applicable
- [ ] I have checked the E2E tests CI results, and my changes do not break them
```

Comment (anomiex) [reply .github/PULL_REQUEST_TEMPLATE.md]: I don't think I like the first-person version. The stated intention here is "nudge", while first person checkboxes come across as much more of a requirement.

**Comment by brbrr** [comment]
Moved the section under the `Changes proposed in this Pull Request`.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (brbrr) [reply .github/PULL_REQUEST_TEMPLATE.md]: Yup, I agree with Brad here, I'd like to keep it as a "nudge". We definitely can iterate on it later as needed.

**Review by jeherve — APPROVED** [review]

**Pull request merged**